src/sidebar/components/test/thread-list-omega-test.js

```diff
@@ -0,0 +1,280 @@
+      fakeMetadata.isHighlight.returns(true);
+      createComponent();
+
+      fakeRootScope.eventCallbacks[events.BEFORE_ANNOTATION_CREATED]({}, {});
+
+      assert.notCalled(fakeStore.clearSelection);
+    });
+
+    it('does not clear the selection in the store if new annotation is a reply', () => {
+      fakeMetadata.isReply.returns(true);
+      createComponent();
+
+      fakeRootScope.eventCallbacks[events.BEFORE_ANNOTATION_CREATED]({}, {});
+
+      assert.notCalled(fakeStore.clearSelection);
+    });
+  });
+
+  context('active scroll to an annotation thread', () => {
+    let stubbedDocument;
+    let stubbedFakeScrollContainer;
+    let fakeScrollTop;
+
+    beforeEach(() => {
+      fakeScrollTop = sinon.stub();
+      stubbedFakeScrollContainer = sinon
+        .stub(fakeScrollContainer, 'scrollTop')
+        .set(fakeScrollTop);
+      stubbedDocument = sinon
+        .stub(document, 'querySelector')
+        .returns(fakeScrollContainer);
+    });
+
+    afterEach(() => {
+      stubbedFakeScrollContainer.restore();
+      stubbedDocument.restore();
+    });
+
+    it('should do nothing if there is no active annotation thread to scroll to', () => {
+      createComponent();
+
+      assert.notCalled(fakeScrollTop);
+    });
+
+    it('should do nothing if the annotation thread to scroll to is not in DOM', () => {
+      createComponent();
+
+      act(() => {
+        fakeRootScope.eventCallbacks[events.BEFORE_ANNOTATION_CREATED](
+          {},
+          { $tag: 'nonexistent' }
+        );
+      });
+
+      assert.notCalled(fakeScrollTop);
+    });
+
+    it('should set the scroll container `scrollTop` to derived position of thread', () => {
+      createComponent();
+
+      act(() => {
+        fakeRootScope.eventCallbacks[events.BEFORE_ANNOTATION_CREATED](
+          {},
+          fakeTopThread.children[3].annotation
+        );
+      });
+
+      // The third thread in a collection of threads at default height (200)
+      // should be at 600px. This setting of `scrollTop` is the only externally-
+      // observable thing that happens here...
+      assert.calledWith(fakeScrollTop, 600);
+    });
+  });
+
+  describe('scroll and resize event handling', () => {
+    const debouncedFn = sinon.stub();
+
+    beforeEach(() => {
+      fakeDebounce.returns(debouncedFn);
+    });
+
+    it('updates scroll position and window height for recalculation on container scroll', () => {
+      createComponent();
+      document
+        .querySelector('.js-thread-list-scroll-root')
+        .dispatchEvent(new Event('scroll'));
+
+      assert.calledOnce(debouncedFn);
+    });
+
+    it('updates scroll position and window height for recalculation on window resize', () => {
+      createComponent();
+      window.dispatchEvent(new Event('resize'));
+
+      // TODO
+      assert.calledTwice(debouncedFn);
+    });
+  });
+
+  context('when top-level threads change', () => {
```

Comment:
I couldn't find a way to test the internal logic of setting new thread heights (https://github.com/hypothesis/client/pull/2044/files#diff-0aa90e2743ab4cd906abb9caa7f3364bR149)

as nothing easily observable happens as a result of that state change.

Reply:
> I couldn't find a way to test the internal logic of setting new thread heights 

For rendered threads, nothing should change, since the value isn't used. If a non-rendered thread height changes, the overall height of the scroll container (ie. the scrollbar range) should change.